### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by war-in**
Closes https://github.com/foundry-rs/starknet-foundry/issues/1588

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,92 @@
 # cairo-profiler
 Profiler for cairo programming language &amp; starknet
 
+## External tools integration
+
+`cairo-profiler` is tool-agnostic which means that it accepts input from any tool. Those tools need to generate
+trace in the [expected](https://github.com/software-mansion/cairo-profiler/blob/d5a5a2722fdf81a0cbabfa44a91cbd69ebe7110d/src/trace_data.rs#L10) format.
```
Comment (war-in) [inline README.md:7]: I'm not sure about this because interface can change in the future. Do you have any suggestions?

File: README.md [hunk 2]
```diff
@@ -1,20 +1,92 @@
...
+## Usage
+
+Usage flow consists of two steps:
+
+- first we need to generate the `profile.pb.gz` file,
+- and then run `pprof` tool.
+
+### Generate pprof file
+
+To generate the file run `cairo-profiler` with the `<PATH_TO_TRACE_DATA>` argument.
```
Comment (MaksymilianDemitraszek) [inline README.md:36]: There is no info about how to get trace data

File: README.md [hunk 3]
```diff
@@ -1,20 +1,92 @@
...
+
+This command will start a web server at the specified port that provides an interactive interface.
+More info [here](https://github.com/google/pprof?tab=readme-ov-file#run-pprof-via-a-web-interface).
+
+## Roadmap
+
+`cairo-profiler` is under active development! Expect a lot of new features to appear soon! 🔥
+
+- [x] Starknet calls profiling
+- [ ] Pure Cairo profiling
```
Comment (MaksymilianDemitraszek) [inline README.md:71]: Function level profiling

File: README.md [hunk 4]
```diff
@@ -1,21 +1,81 @@
...
 
-## Reading the profiling results
+## External tools integration
+
+`cairo-profiler` is a tool-agnostic profiler which means that it accepts input from any tool. Those tools need to generate
+trace in the [expected](https://github.com/software-mansion/cairo-profiler/blob/main/src/trace_data.rs) format.
+
+### Integrated tools
+
+- [x] [Starknet Foundry](https://github.com/foundry-rs/starknet-foundry) - check how to generate the input file [here](https://foundry-rs.github.io/starknet-foundry/testing/profiling.html)
```
Comment (piotmag769) [inline README.md:26]: The link will work tomorrow xD (added docs today)

File: README.md [hunk 5]
```diff
@@ -1,21 +1,128 @@
 # cairo-profiler
-Profiler for cairo programming language &amp; starknet
 
-## Setup for development
+Profiler for Cairo programming language &amp; Starknet.
```
Comment (war-in) [inline README.md:3]: Would be nice to add some `.gif` showing how the profiled trace looks like

File: README.md [hunk 6]
```diff
@@ -1,21 +1,128 @@
...
 
+```shell
+curl -L https://raw.githubusercontent.com/software-mansion/cairo-profiler/main/scripts/install.sh | sh
 ```
-cargo run
+
+If you want to install a specific a version, run the following command with the requested version:
+
+```shell
+curl -L https://raw.githubusercontent.com/software-mansion/cairo-profiler/main/scripts/install.sh | sh -s -- v0.1.0
```
Comment (war-in) [inline README.md:16]: We need one more `if` (for the Windows users 💀) because the installation script will not work on windows

**Review by piotmag769 — APPROVED** [review]

**Pull request merged**